src/authentication/authentication.service.ts

```diff
@@ -1,17 +1,47 @@
 import { Injectable } from '@nestjs/common';
 import { LoginDto, RegisterDto, ResetPasswordDto } from './dto';
+import { supabase } from '../config/supabase.client';
 
 @Injectable()
 export class AuthenticationService {
-  async register(registerDto: RegisterDto): Promise<any> {
-    // Implement registration logic
+  async register(registerDto: RegisterDto): Promise<{ data: any; error: boolean; message?: string }> {
+    try {
+      const { email, password } = registerDto;
+      const { data: user, error } = await supabase.auth.signUp({
+        email,
+        password,
+      });
+      if (error) {
+        if (error.message.includes('Email rate limit exceeded')) {
```

Comment:
We can work this as a thrown error instead of returning the error as a property.